### ISSUE TITLE
Fails to compile with GCC 5.0: narrowing conversion in koepac_voice_archive.cc

### ISSUE THREAD

**Issue by jbeich** [eglaysher/rlvm#65]
`-Wno-narrowing` doesn't help to avoid the following error  [found](http://muscles.dragonflybsd.org/bulk/latest-per-pkg/rlvm/0.14.0.20150223_1/bleeding-edge-potential.log) via DPorts. Maybe use `int16_t` instead.
```
src/systems/base/koepac_voice_archive.cc:142:27: error: narrowing conversion of '255
' from 'int' to 'int8_t {aka signed char}' inside { }
     0x7e, 0x81, 0x7f, 0x80};
                           ^
src/systems/base/koepac_voice_archive.cc:142:27: error: narrowing conversion of '25$
' from 'int' to 'int8_t {aka signed char}' inside { }
src/systems/base/koepac_voice_archive.cc:142:27: error: narrowing conversion of '25$
' from 'int' to 'int8_t {aka signed char}' inside { }
...
```

**Comment by minirop** [comment]
if it does not go over 255, uint8_t shouldn't be prefered?

**Comment by eglaysher** [comment]
uint8_t probably should be prefered; given some of the libraries that rlvm links in, I wouldn't expect rlvm to compile with signed characters. (Most of the code which assumes characters are always unsigned predates rlvm by years. Like most of the image and audio parsing code, the koepac decoding code originally comes from xclannad.)

I suspect a switch to uint8_t...might work here. But now I'm wondering exactly why the variable d (line 213) is signed. That might be significant... Every time I've touched some of these internals, I've made a hash of it...
